lib/ddtrace/method_tracer.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+module Datadog
+  # \MethodTracer adds helpers for unobtrusive methods tracking.
+  #
+  # == Example
+  #
+  #   Datadog::MethodTracer.trace_methods(ExampleClass, :some_instance_method, :other_instance_method)
+  #   Datadog::MethodTracer.trace_singleton_methods(ExampleClass, :some_class_method, :other_class_method)
+  #
+  module MethodTracer
+    module_function
+
+    def trace_methods(klass, *method_names)
+      instrumentation.trace_methods klass, *method_names
+    end
+
+    def trace_singleton_methods(klass, *method_names)
+      instrumentation.trace_singleton_methods klass, *method_names
+    end
+
+    def instrumentation
+      @_instrumentation ||= Instrumentation.new
```

Comment:
It's good to give a hint to users that this environment variable present in their application comes from Datadog:
```suggestion
      @dd_instrumentation ||= Instrumentation.new
```